src/io/delimited.cr

```diff
@@ -1,124 +1,120 @@
-module IO
-  # An `IO` that wraps another `IO`, and only reads up to the beginning of a
-  # specified delimiter.
-  #
-  # This is useful for exposing part of an underlying stream to a client.
-  #
-  # ```
-  # io = IO::Memory.new "abc||123"
-  # delimited = IO::Delimited.new(io, read_delimiter: "||")
-  #
-  # delimited.gets_to_end # => "abc"
-  # delimited.gets_to_end # => ""
-  # io.gets_to_end        # => "123"
-  # ```
-  class Delimited
-    include IO
-
-    # If `#sync_close?` is `true`, closing this `IO` will close the underlying `IO`.
-    property? sync_close
-
-    getter read_delimiter
-    getter? closed : Bool
-
-    @delimiter_buffer : Bytes
-    @active_delimiter_buffer : Bytes
-
-    # Creates a new `IO::Delimited` which wraps *io*, and can read until the
-    # byte sequence *read_delimiter* (interpreted as UTF-8) is found. If
-    # *sync_close* is set, calling `#close` calls `#close` on the underlying
-    # `IO`.
-    def self.new(io : IO, read_delimiter : String, sync_close : Bool = false)
-      new(io, read_delimiter.to_slice, sync_close)
-    end
+# An `IO` that wraps another `IO`, and only reads up to the beginning of a
+# specified delimiter.
+#
+# This is useful for exposing part of an underlying stream to a client.
+#
+# ```
+# io = IO::Memory.new "abc||123"
+# delimited = IO::Delimited.new(io, read_delimiter: "||")
+#
+# delimited.gets_to_end # => "abc"
+# delimited.gets_to_end # => ""
+# io.gets_to_end        # => "123"
+# ```
+class IO::Delimited < IO
+  # If `#sync_close?` is `true`, closing this `IO` will close the underlying `IO`.
+  property? sync_close
+
+  getter read_delimiter
+  getter? closed : Bool
+
+  @delimiter_buffer : Bytes
+  @active_delimiter_buffer : Bytes
+
+  # Creates a new `IO::Delimited` which wraps *io*, and can read until the
+  # byte sequence *read_delimiter* (interpreted as UTF-8) is found. If
```

Comment:
shouldn't the delimiter be interpreted in the encoding of the underling io?

Reply:
Maybe, but we can discuss that in a separate PR :-)